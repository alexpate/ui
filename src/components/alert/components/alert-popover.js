/* @flow */
import React from 'react';
import {Box} from '../../layout';

type TProps = {
  children: React$Element<*>,
  topOffset?: number,
};

export default function AlertPopover(props: TProps) {
  const {children, topOffset = 0} = props;

  return (
    <Box position="fixed" right={32} top={topOffset + 16} maxWidth={400}>
      {React.Children.map(children, child =>
        <Box marginBottom={8}>
          {React.cloneElement(child, {justifyContent: 'left'})}
        </Box>
      )}
    </Box>
  );
}
