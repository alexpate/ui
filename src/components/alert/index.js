/* @flow */
export {default} from './alert';
export {default as AlertPopover} from './components/alert-popover';
