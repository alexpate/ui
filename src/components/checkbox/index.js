/* @flow */
export {default} from './checkbox';
