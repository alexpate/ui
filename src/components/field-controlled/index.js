/* @flow */
export {default} from './field-controlled';
