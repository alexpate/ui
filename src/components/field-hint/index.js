export {default as FieldHint} from './field-hint';
