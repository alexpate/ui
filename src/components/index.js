/**
 * Typography
 */
export { H1 } from './typography/h1';
export { H2 } from './typography/h2';
export { H3 } from './typography/h3';
export { H4 } from './typography/h4';
export { Text } from './typography/text';
export { Heading } from './typography/heading';

export { Button } from './button';
export { ButtonGroup } from './button-group';
export { FieldHint } from './field-hint';
export { Icon } from './icon';
export { Input } from './input';
export { List } from './list';
export { LoadingSpinner } from './loading-spinner';
export { Radio } from './radio';
export { StarRating } from './star-rating';
export { StepMarks } from './step-marks';
export { ToggleButton } from './toggle-button';
