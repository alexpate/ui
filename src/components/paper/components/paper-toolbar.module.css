.ui-paper-toolbar {
  display: flex;
  border-bottom: 1px solid var(--grey100);
  position: relative;
  flex-direction: column;
  align-items: stretch;
  flex-shrink: 0;
  align-content: flex-start;
  padding: 0 2.4rem;
}
