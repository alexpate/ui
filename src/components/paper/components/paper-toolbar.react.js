/* @flow */
import * as React from 'react';
import cx from 'classnames';

import {UIBase} from '../../layout';

import styles from './paper-toolbar.css';

type TProps = {
  children?: React.Node,
  className?: string | Object,
};

export function PaperToolbar(props: TProps) {
  const {children, className} = props;

  const _classNames = cx(
    {
      [styles['ui-paper-toolbar']]: true,
    },
    className
  );

  return <UIBase className={_classNames}>{children}</UIBase>;
}
