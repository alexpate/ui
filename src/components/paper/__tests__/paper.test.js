/* @flow */
/* eslint-env jest */
import * as React from 'react';
import {sheet} from 'emotion';
import serializer from 'jest-glamor-react';
import renderer from 'react-test-renderer';

import Paper from 'components/paper';
import {PaperToolbar} from '../components/paper-toolbar';

expect.addSnapshotSerializer(serializer(sheet));

describe('components/paper', () => {
  describe('Paper', () => {
    const defaultProps = {};
    const create = (props = {spacing: true}) =>
      renderer
        .create(
          <Paper {...defaultProps} {...props}>
            Paper Contents
          </Paper>
        )
        .toJSON();

    test('should render shallow component ok', () => {
      const element = create();
      expect(element).toMatchSnapshot();
    });

    test('should render a component with an elevation of 3', () => {
      const element = create({
        elevation: 3,
      });
      expect(element).toMatchSnapshot();
    });
  });

  describe('PaperToolbar', () => {
    test('should render shallow component ok', () => {
      const element = renderer.create(
        <PaperToolbar>
          <Paper>Toolbar child contents</Paper>
        </PaperToolbar>
      );
      expect(element).toMatchSnapshot();
    });
  });
});
