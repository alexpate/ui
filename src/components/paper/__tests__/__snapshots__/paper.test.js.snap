// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/paper Paper should render a component with an elevation of 3 1`] = `
.glamor-0 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="root level-3 hover-level-1 border rounded glamor-0"
  onClick={undefined}
  style={Object {}}
>
  Paper Contents
</div>
`;

exports[`components/paper Paper should render shallow component ok 1`] = `
.glamor-0 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="root level-1 hover-level-1 border rounded glamor-0"
  onClick={undefined}
  spacing={true}
  style={Object {}}
>
  Paper Contents
</div>
`;

exports[`components/paper PaperToolbar should render shallow component ok 1`] = `
.glamor-0 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.glamor-1 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0px 25px;
}

<div
  className="root glamor-1"
  style={Object {}}
>
  .glamor-0 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
    className="root level-1 hover-level-1 border rounded glamor-0"
    onClick={undefined}
    style={Object {}}
  >
    Toolbar child contents
  </div>
</div>
`;
